Code/CSharpAdvanced/FlowerExtensions.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CSharpAdvanced
+{
+    public static class FlowerExtensions
+    {       
+        public static string GenerateRandomColor(this Flower flower)
+        {
+            Random random = new Random();
+            string[] colors = { "Red", "Blue", "Yellow", "Pink", "Purple" };
+            int randomIndex = random.Next(colors.Length);
+            return colors[randomIndex];
+        }
+        public static void PrintFlowerInfo(this Flower flower)
+        {
+            Console.WriteLine($"Flower Name is {flower.Name}");
+            Console.WriteLine($"Flower Color is {flower.Color}");
+            Console.WriteLine($"Flower Price is {flower.Price}");
+            Console.WriteLine($"Flower Kind is {flower.GetType().Name}");
+        }
+        public static bool IsPriceExpensive(this Flower flower)
+        {
+            return flower.Price > 50;
```

Comment:
Do not hardcode values, use variables instead 😊

Reply:
done